lib/http3/server.c

```diff
@@ -27,6 +27,36 @@
 #include "h2o/http3_internal.h"
 #include "./../probes_.h"
 
+/**
+ * the scheduler
+ */
+struct st_h2o_http3_req_scheduler_t {
+    struct {
+        struct {
+            h2o_linklist_t high;
+            h2o_linklist_t low;
+        } urgencies[H2O_ABSPRIO_NUM_URGENCY_LEVELS];
+        size_t smallest_urgency;
+    } active;
+    h2o_linklist_t conn_blocked;
+};
+
+/**
+ *
+ */
+struct st_h2o_http3_req_scheduler_node_t {
+    h2o_linklist_t link;
+    h2o_absprio_t priority;
+    uint64_t call_cnt;
+};
+
+/**
+ * callback used to compare precedence of the entries within the same urgency level (e.g., by comparing stream IDs)
+ */
+typedef int (*h2o_http3_req_scheduler_compare_cb)(struct st_h2o_http3_req_scheduler_t *sched,
+                                                  struct st_h2o_http3_req_scheduler_node_t *x,
+                                                  struct st_h2o_http3_req_scheduler_node_t *y);
+
```

Comment:
Maybe a nitpick, but would there be a use case where `x, y` must be non-const? I feel like they can be `const` pointers (and possibly `sched` as well?).

Reply:
Good point. Added `const` to x and y. `sched` is left without `const` due to the two reasons:
* the comparator of `qsort_r` does not have `const` for the thunk
* our code base is loose in specifying when and what is const